modules/Bio/EnsEMBL/Compara/RunnableDB/SpeciesTree/CheckSketches.pm

```diff
@@ -169,19 +174,19 @@ sub _check_for_distance_files {
 	my $dir = $self->param_required('sketch_dir');
 	my $collection = $self->param_required('collection');
 	my @dist_files = glob "$dir/*$collection*.dists";
-	my @needed_gdb_ids = map {$_->dbID} @{ $self->param('genome_dbs') };
+	my $gdb_id_map = { map {$_->_get_genome_dump_path($self->param('genome_dumps_dir')) => $_->dbID} @{ $self->param('genome_dbs') } };
 
 	my $gdb_adaptor = $self->compara_dba->get_GenomeDBAdaptor;
 
 	foreach my $dfile ( @dist_files ) {
 		print " --- checking $dfile\n";
 		my $dist_matrix = Bio::EnsEMBL::Compara::Utils::DistanceMatrix->new( -file => $dfile );
-		$dist_matrix = $dist_matrix->convert_to_genome_db_ids($gdb_adaptor);
-
-		my @matrix_members = $dist_matrix->members;
-		my $overlap = $self->_overlap(\@needed_gdb_ids, \@matrix_members);
-		print " --- --- overlap = $overlap (" . scalar @needed_gdb_ids . " needed)\n";
-		return $dfile if $overlap >= scalar @needed_gdb_ids;
+		try {
+			$dist_matrix = $dist_matrix->filter_and_convert_genome_db_ids($gdb_id_map);
+			return $dfile;
+		} catch {
+			warn "Caught error whilst parsing distance matrix file $dfile\nError: $_";
+		}
```

Comment:
eHive has built in warnings that we use : `$self->warning()` - this means the warning will be appropriately captured and stored in the database - so we should stick to this.
```suggestion
        if ( -e $dfile ) {
            $dist_matrix = $dist_matrix->filter_and_convert_genome_db_ids($gdb_id_map);
            return $dfile;
        }
        else {
            $self->warning("Could not parse distance matrix file $dfile");
        }
```
(Excuse the indentation in this suggestion - it probably doesn't fit)

Reply:
The error here doesn't occur when the `$dfile` doesn't exist (I mean, I guess one would, but the way we gather them they should), but rather when `filter_and_convert_genome_db_ids` dies because an expected genome_db_id is missing. Therefore I don't believe you can rely on `-e $dfile`. 

This wasn't required before because the previous code here had a different method of checking if all the expected IDs were in the distance matrix.  `filter_and_convert_genome_db_ids` now just dies if this is the case.

Reply:
When `filter_and_convert_genome_db_ids` dies that should be enough. eHive will capture that. Additionally you could include a `$self->warning()` in `filter_and_convert_genome_db_ids` before it dies.
I meant to put `if ( -s $dfile ) {` rather than `-e` - but had only had one coffee.

Reply:
But we don't want the whole process to die if `filter_and_convert` dies. That step is supposed to loop through all prior/current dist files and look for one which already covers all the genome_dbs we need. The `$dfile` come from a glob so should always exist (technically you have a race condition there but oh well...). Maybe `filter_and_convert` should return `undef` if there are missing db IDs but my python mindset is to make stuff fail at the point of error and deal with it when you need to, not to mysteriously pass around null values as a proxy for failure

Reply:
Drop the python mindset then, this is Perl. Return `undef` for missing gdbs. You have over-complicated this.

Reply:
Changes applied in later commits.